io-engine/src/core/env.rs

```diff
@@ -466,6 +468,9 @@ async fn do_shutdown(arg: *mut c_void) {
         spdk_rpc_finish();
         spdk_subsystem_fini(Some(reactors_stop), arg);
     }
+    MayastorEnvironment::global_or_default()
```

Comment:
Shouldn't this be generated at the start of shutdown?

Reply:
Added the event at the start.